### PR TITLE
Update Android Build Settings

### PR DESCRIPTION
**Pull request by cacheung**
Update Gradle Plugin to 7.3.1
Update Gradle Version to 7.5
Update Java Version to Version_1_8
Update Gradle dependencies to the latest versions

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/edgeconsent/build.gradle [hunk 1]
```diff
@@ -80,18 +81,21 @@ android {
 android.libraryVariants.all { variant ->
     tasks.withType(Javadoc) {
         source = [android.sourceSets.main.java.sourceFiles, android.sourceSets.phone.java.sourceFiles]
+
         ext.androidJar = "${android.sdkDirectory}/platforms/${android.compileSdkVersion}/android.jar"
 
         doFirst{classpath = files(variant.javaCompile.classpath.files) + files(ext.androidJar)}
 
+        failOnError false
+
```
Comment (cacheung) [inline code/edgeconsent/build.gradle:90]: Add failOnError to bypass the javadoc android.jar components not found error.

File: code/edgeconsent/build.gradle [hunk 2]
```diff
@@ -243,31 +214,35 @@ task platformFunctionalTestJacocoReport(type: JacocoReport, dependsOn: "createPh
     ])
 
     reports {
-        xml.enabled true
-        csv.enabled false
-        html.enabled true
+        xml.required = true
```
Comment (cacheung) [inline code/edgeconsent/build.gradle:217]: enabled will be deprecated in gradle 8, replaced with required api.

File: code/app/build.gradle [hunk 3]
```diff
@@ -35,8 +38,8 @@ android {
 
 dependencies {
     implementation fileTree(dir: "libs", include: ["*.jar"])
```
Comment (emdobrin) [inline code/app/build.gradle:40]: we can remove this too

File: code/app/build.gradle [hunk 4]
```diff
@@ -14,15 +16,16 @@ spotless {
 }
 
 android {
-    compileSdkVersion 30
-    buildToolsVersion "30.0.2"
+      namespace "com.adobe.marketing.mobile.consentTestApp"
+      compileSdk rootProject.ext.compileSdkVersion
+      buildToolsVersion rootProject.ext.buildToolsVersion
```
Comment (emdobrin) [inline code/app/build.gradle:21]: buildToolsVersion should no longer be required with latest versions of agp - https://developer.android.com/studio/releases/gradle-plugin.html#behavior_changes_1

File: code/build.gradle [hunk 5]
```diff
@@ -1,53 +1,34 @@
...
-
-    apply plugin: 'maven-publish'
-}
-
-task clean(type: Delete) {
-    delete rootProject.buildDir
+plugins {
+    // Android Gradle Plugin version
+    id 'com.android.application' version '7.3.1' apply false
+    id 'com.android.library' version '7.3.1' apply false
```
Comment (emdobrin) [inline code/build.gradle:5]: can we use 7.3.0 for all edge extensions?

File: code/build.gradle [hunk 6]
```diff
@@ -1,53 +1,34 @@
...
-    targetCompatibility = JavaVersion.VERSION_1_7
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
 
     // dependencies
-    junitVersion = "1.1.2"
+    junitVersion = "1.1.3"
     mockitoCoreVersion = "2.28.2"
-    buildToolsVersion = "29.0.3"
+    buildToolsVersion = "30.0.3"
```
Comment (emdobrin) [inline code/build.gradle:23]: same comment here, this should no longer be required. do you see a warning when using this value?

File: code/edgeconsent/build.gradle [hunk 7]
```diff
@@ -46,7 +48,6 @@ android {
     buildTypes {
         debug {
             testCoverageEnabled true
-            debuggable true
```
Comment (emdobrin) [inline code/edgeconsent/build.gradle:49]: we should keep this around for debug builds

File: code/edgeconsent/build.gradle [hunk 8]
```diff
@@ -174,12 +141,12 @@ publishing {
...
-                    connection = 'scm:git:github.com//Adobe-Marketing-Cloud/acp-sdks.git'
-                    developerConnection = 'scm:git:ssh://github.com//Adobe-Marketing-Cloud/acp-sdks.git'
-                    url = 'https://github.com/Adobe-Marketing-Cloud/acp-sdks'
+                    connection = 'scm:git:github.com//adobe/aepsdk-edgeconsent-android.git'
+                    developerConnection = 'scm:git:ssh://github.com//adobe/aepsdk-edgeconsent-android.git'
+                    url = 'https://github.com/adobe/aepsdk-edgeconsent-android'
                 }
                 withXml {
-                    def dependenciesNode = asNode().appendNode('dependencies')
+                    def dependenciesNode = asNode.appendNode('dependencies')
```
Comment (emdobrin) [inline code/edgeconsent/build.gradle:149]: I believe this warning was just saying asNode() uses default null string which did not seem like a problem, does asNode return same object here?

File: code/edgeconsent/build.gradle [hunk 9]
```diff
@@ -222,12 +189,12 @@ task platformUnitTestJacocoReport(type: JacocoReport, dependsOn: "testPhoneDebug
     additionalClassDirs.setFrom files([debugTree])
```
Comment (emdobrin) [inline code/edgeconsent/build.gradle:189]: If there are no ADB prefixed classes for this repo source code, we can remove that from excludeRegex (same for platformFunctionalTestJacocoReport)

File: code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/Consents.java [hunk 10]
```diff
@@ -185,7 +185,7 @@ public boolean equals(final Object comparingConsentObject) {
 
 	/**
 	 * Compares the current consent instance the with the passed object ignoring the timestamp field in metadata
-	 *
+	 * @param comparingConsent the current consents instance
```
Comment (emdobrin) [inline code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/Consents.java:188]: the current consents instance is "this", while comparingConsent is the new consent object to be compared against current consent settings

File: code/edgeconsent/build.gradle (hunk 9 above)
Comment (cacheung) [reply code/edgeconsent/build.gradle:189]: We have ADBCountDownLatch class.

Comment (kevinlind) [reply code/edgeconsent/build.gradle:189]: The ADBCountDownLatch class is a utility class used for testing and is not included in the coverage report. There should not be any "ADB" classes in the main (non-test) source set. Can you remove the "ADB" exclusion from the coverage configuration?

File: code/edgeconsent/build.gradle [hunk 11]
```diff
@@ -243,31 +210,35 @@ task platformFunctionalTestJacocoReport(type: JacocoReport, dependsOn: "createPh
...
     androidTestImplementation "androidx.test.ext:junit:${rootProject.ext.junitVersion}"
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
     // Using older version of PowerMock as it supports minimum Android API < 26
     // the root issue appears to be a dependency with Objenesis in Mockito-Core
     // where Objenesis 2 requires minimum Android API 26
     androidTestImplementation 'org.powermock:powermock-module-junit4:1+'
-    androidTestImplementation 'com.fasterxml.jackson.core:jackson-databind:2.10.5'
+    androidTestImplementation 'com.fasterxml.jackson.core:jackson-databind:2.12.7'
+    androidTestImplementation project(path: ':edgeconsent')
```
Comment (kevinlind) [inline code/edgeconsent/build.gradle:241]: It shouldn't be required to explicitly include the project module for the instrumented tests. Can you remove this line?

File: code/gradle.properties [hunk 12]
```diff
@@ -15,4 +15,6 @@ mavenRepoDescription=Adobe Experience Platform Edge Consent Collection extension
 mavenUploadDryRunFlag=false
 
 # production versions for production build
-mavenCoreVersion=1.8.0
\ No newline at end of file
+mavenCoreVersion=1.8.0
+
+android.disableAutomaticComponentCreation=true
```
Comment (kevinlind) [inline code/gradle.properties:20]: nit: move this up with the other "android" configuration options so they are grouped together.

File: code/app/build.gradle [hunk 13]
```diff
@@ -1,5 +1,7 @@
-apply plugin: 'com.android.application'
-apply plugin: "com.diffplug.spotless"
+plugins {
+    id 'com.android.application'
+    id 'com.diffplug.spotless'
+}
 
 spotless {
```
Comment (emdobrin) [inline code/app/build.gradle:6]: We can take this in a separate PR, but can we define these spotless settings only once for all sources (source code, test app, tests)? 
cc @kevinlind

**Review by emdobrin — COMMENTED** [review]
Looks good @cacheung with couple open questions.

File: code/app/build.gradle (hunk 13 above)
Comment (kevinlind) [reply code/app/build.gradle:6]: One possible option is to create a common Gradle file for the Spotless configuration and apply it to the other build.gradle files. Here is an example of Core's Checkstyle configuration https://github.com/adobe/aepsdk-core-android/blob/dev-v2.0.0/code/checkstyle.gradle. This file is then included using `apply from: '../checkstyle.gradle'`.
For some of our extensions, such as Identity and Bridge, which have Kotlin test apps, the Spotless configurations are different so a common Gradle file would not help much.

**Review by kevinlind — APPROVED** [review]

**Review by emdobrin — APPROVED** [review]

**Pull request merged**